### PR TITLE
Update to Fontawesome 5

### LINKED ISSUE #402: Upgrade to Font Awesome v5 (Free)
Currently, we are using Font Awesome v4 (or a custom subset with [Fontello](https://fontello.com)). It would be good to upgrade to Font Awesome v5 (Free).

Unfortunately, Fontello does not offer customization for FA v5. There are payed services like [Font Awesome Kits](https://blog.fontawesome.com/introducing-font-awesome-kits/) [Fort Awesome](https://fortawesome.com/start) that allows customization.

- [ ] Research if there are other libraries or services available to create a subset of FA v5 (Free); Otherwise we need to use the whole free icons set
- [ ] Follow the [FA v5 upgrade guidelines](https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4) -> we can use the (free) _solid_ set, but some icon names have changed

### PR DESCRIPTION
**Pull request by sgratzl**
closes #402 , based on #477 

**prerequisites**:

- [x] branch is up-to-date with the branch to be merged with, i.e. develop
- [x] build is successful
- [x] code is cleaned up and formatted

## Summary

updates to Fontawesome 5 free as good as it gets

### CODE REVIEW

File: src/styles/_vars.scss [hunk 1]
```diff
@@ -79,6 +79,11 @@ $lu_aggregate_hierarchy_color: #000 !default;
 $lu_taggle_border_color: #6d6c6c !default;
 
 $lu_assets: './assets' !default;
+
+// $fa-font-path: '~@fortawesome/fontawesome-free/webfonts' !default;
+// @import '~@fortawesome/fontawesome-free/scss/solid';
+// @import '~@fortawesome/fontawesome-free/scss/regular';
+// $lu_use_font_awesome: true !default;
 $lu_use_font_awesome: false !default;
```
Comment (thinkh) [inline src/styles/_vars.scss:87]: Do I understand this PR correctly, that by default LineUp is still using Fontawesome 4 (via custom lu-font)? And if somebody wants to have FontAwesome 5 he/she must change the flag and add the imports as described in the comments?

**Review by thinkh — COMMENTED** [review]
The netlify preview shows still the FontAwesome 4. Is this intentional? Why don't we make FontAwesome 5 the default and drop FA 4?

**Review by thinkh — CHANGES_REQUESTED** [review]
I'm testing this PR now locally and I'm able to see the FA 5 icons after uncommenting the respective lines in _vars.scss.

Can we keep the column type icons also with FA 5? 

**Before (with FA 4)**

grafik

**After (with FA 5)**

grafik

[The icons are also available as SVG in the assets folder](https://github.com/lineupjs/lineupjs/tree/sgratzl/fontawesome5/src/assets). So we can either use the SVG or create a custom font with only the icons (instead of the ones included in the lu-font).

**Comment by thinkh** [comment]
I cannot figure out how to use FA 5 outside of LineUp with the current PR state. 

1. Checkout this PR
2. `yarn run build:dev`

The `build` contains only the `LineUpJS.css` file, but no SASS file.

grafik

Importing the `LineUpJS.css` file in a different app/project still shows the FA 4 icons.

In my opinion we have now the following options:

1. Create a custom build with `$lu_use_font_awesome: true` -> probably not the way to go when using LineUp as a library
2. Generate additional SCSS files that can imported in a different app/project after setting `$lu_use_font_awesome: true`
3. Use FA 5 as LineUp default icons

@sgratzl Could you please generate the SCSS files or use FA 5 as default icons? Thanks.

**Comment by thinkh** [comment]
> I updated the custom font also to use Font Awesome 5 (see README notes about using [fontello/fontello#665](https://github.com/fontello/fontello/pull/665)).

Thanks, that looks good after switching the flag `$lu_use_font_awesome` to true.

> I guess so far the method is to build the style sheet from source directly. Part of the package should be the src/styles.scss which you can import in your setup and change variables before. (similar what tdp_core is doing)

Yes, true that would be the solution. Unfortunately, it does not work for us in tdp_core, due to the different sass compilers (LineUp = Dart Sass; tdp_core = Node Sass):

grafik

**Comment by sgratzl** [comment]
I added a built artifact: `LineUpJS_fontawesome.css` which is a style sheet version with the font awesome flag enabled. Moreover, I disabled by default that the Font awesome files are bundled, so one has to provide them manually, see also `demo/fontawesome.html`

**Comment by sgratzl** [comment]
@thinkh ping

**Review by thinkh — APPROVED** [review]

**Pull request merged**